datafusion/optimizer/src/push_down_filter.rs

```diff
@@ -530,14 +530,14 @@ fn push_down_join(
     parent_predicate: Option<&Expr>,
 ) -> Result<Transformed<LogicalPlan>> {
     // Split the parent predicate into individual conjunctive parts.
-    let predicates = parent_predicate
-        .map_or_else(Vec::new, |pred| split_conjunction_owned(pred.clone()));
+    let predicates =
```

Comment:
this certainly looks nicer but it seems like the expressions are still cloned at the end. Though maybe in this case only the conjunctions are cloned (not the BinaryExprs with AND) so it would be an improvement